hil/ext/switches/brocade.py

```diff
@@ -302,4 +303,5 @@ def _make_request(self, method, url, data=None,
         if r.status_code >= 400 and \
            r.status_code not in acceptable_error_codes:
             logger.error('Bad Request to switch. Response: %s', r.text)
+            raise SwitchError('Bad Request to switch')
```

Comment:
I called `_get_native_vlan` on a non existing port to raise the error. 
And this was the result:
```
(Pdb) p r.status_code
404
(Pdb) p r.text
u''
(Pdb) p r.reason
'Not Found'
(Pdb) dir(r)
['__attrs__', '__bool__', '__class__', '__delattr__', '__dict__', '__doc__', '__enter__', '__exit__', '__format__', '__getattribute__', '__getstate__', '__hash__', '__init__', '__iter__', '__module__', '__new__', '__nonzero__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setstate__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_content', '_content_consumed', '_next', 'apparent_encoding', 'close', 'connection', 'content', 'cookies', 'elapsed', 'encoding', 'headers', 'history', 'is_permanent_redirect', 'is_redirect', 'iter_content', 'iter_lines', 'json', 'links', 'next', 'ok', 'raise_for_status', 'raw', 'reason', 'request', 'status_code', 'text', 'url']
```
So I think it might be a good idea to log `r.reason` in addition to `r.text` so that we have enough information to work with when debugging. I think `method` and `url` might be useful too.

Reply:
sounds great! I will use `r.reason`